node/src/Commands.ts

```diff
@@ -1476,7 +1480,18 @@ export function createZAdd(
         args.push("INCR");
     }
 
-    membersAndScores.forEach((p) => args.push(p.score.toString(), p.element));
+    if (Array.isArray(membersAndScores)) {
+        for (const memberAndScore of membersAndScores) {
+            args.push(memberAndScore.score.toString(), memberAndScore.element);
+        }
+    } else {
+        const members = Object.keys(membersAndScores);
+
+        for (const member of members) {
+            args.push(membersAndScores[member].toString(), member);
+        }
+    }
+
```

Comment:
```suggestion
    if (Array.isArray(membersAndScores)) {
        for (let i = 0, len = membersAndScores.length; i < len; i++) {
            const item = membersAndScores[i];
            args.push(item.score.toString(), item.element);
        }
    } else {
        const members = Object.keys(membersAndScores);
        for (let i = 0, len = members.length; i < len; i++) {
            const member = members[i];
            args.push(membersAndScores[member].toString(), member);
        }
    }

```

` for( let i ...)` is avoiding iterator creation.
Using local `member` and `item` minimizing the object property lookup.